pkg/kubeapiserver/admission/configuration/configuration_manager.go

```diff
@@ -81,9 +93,25 @@ func (a *poller) notReady() {
 	a.ready = false
 }
 
+func (a *poller) timeoutBootstrap() {
+	// bootstrapGracePeriod is read-only, so no lock is required
+	timer := time.NewTimer(a.bootstrapGracePeriod)
+	go func() {
+		<-timer.C
+		a.lock.Lock()
+		defer a.lock.Unlock()
+		a.bootstrapped = true
+		a.bootstrappedCondition.Broadcast()
+	}()
+}
+
 func (a *poller) configuration() (runtime.Object, error) {
-	a.lock.RLock()
-	defer a.lock.RUnlock()
+	a.once.Do(a.timeoutBootstrap)
+	a.lock.Lock()
+	defer a.lock.Unlock()
+	if !a.bootstrapped {
+		a.bootstrappedCondition.Wait()
```

Comment:
If initializers are disabled, it wouldn't be called.  I think my worry is the difference between this code:

```
# hold and then release

|-------|
   |----| 
      |-|
        ^ failure
             | <- fails instantly
```

and

```
# each one individually
|--------| <- fail and return to client
     |--------| <- fail and return to client
              |--------| <- fail and return to client
```